frontend/lib/views/truck_direction_map_view.dart

```diff
@@ -42,11 +46,17 @@ class MapDirectionState extends State<MapDirectionView> {
 	BitmapDescriptor destinationIcon;
 	LatLng _sourceLocation;
 	LatLng _destLocation;
+
+
 	@override 
 	void initState() {
```

Comment:
We might need to move async function out of initState(), and use a boolean flag to indicate whether the required data has been loaded or not. Currently if the `userLocation` is not set in the callback but definition of `_sourceLocation` refers to it, null value reference value will be thrown.